### PR TITLE
Use better typing and strum in CLI

### PR DESCRIPTION
**Pull request by DaughterOfMars**
# Description of change

I split these changes out of the CLI delegation PR for easier reviewing

### CODE REVIEW

File: cli/src/wallet_cli/completer.rs [hunk 1]
```diff
@@ -7,52 +7,9 @@ use colored::Colorize;
...
-    "vote",
-    "stop-participating",
-    "participation-overview",
-    "voting-power",
-    "increase-voting-power",
-    "decrease-voting-power",
-    "voting-output",
-    "help",
-];
+use crate::wallet_cli::WalletCommand;
```
Comment (thibault-martinez) [inline cli/src/wallet_cli/completer.rs:12]: Damn :o

File: cli/src/wallet_cli/completer.rs [hunk 2]
```diff
@@ -68,7 +25,7 @@ impl Completer for WalletCommandCompleter {
     ) -> rustyline::Result<(usize, Vec<Self::Candidate>)> {
         Ok((
             0,
-            WALLET_COMMANDS
+            WalletCommand::VARIANTS
```
Comment (thibault-martinez) [inline cli/src/wallet_cli/completer.rs:28]: That's defined by `strum`?

Comment (thibault-martinez) [reply cli/src/wallet_cli/completer.rs:28]: Had a look at it, pretty cool

File: cli/src/wallet_cli/completer.rs [hunk 3]
```diff
@@ -7,52 +7,9 @@ use colored::Colorize;
...
-    "output",
-    "outputs",
-    "send",
-    "send-native-token",
-    "send-nft",
-    "sync",
-    "transaction",
-    "transactions",
-    "tx",
-    "txs",
```
Comment (thibault-martinez) [inline cli/src/wallet_cli/completer.rs:45]: This used to contain shortcuts though, but I'm not too bothered removing them

**Review by Thoralf-M — APPROVED** [review]

**Review by thibault-martinez — APPROVED** [review]

File: cli/src/wallet_cli/completer.rs (hunk 3 above)
Comment (Thoralf-M) [reply cli/src/wallet_cli/completer.rs:45]: Could be added to the iterator part in the completer if we want it

**Review by Alex6323 — APPROVED** [review]

**Pull request merged**